### PR TITLE
Cleanspeak integration (frontend)

### PR DESCRIPTION
**Pull request by lualparedes**
Changes introduced:

- The `sendMessage()` method of the `Chat` component was modified to make a call to the [new moderation end point](https://github.com/UPchieve/server/pull/24) to determine if the message is clean, and modify the UI accordingly:
  - *Message is clean*: new message is printed (just as usual)
  - *Message not clean*: new message isn't printed, and a warning is shown
  - *The backend coundn't determine whether the message is clean*: new message gets printed to avoid breaking the UX and the error returned is logged to the console
- A new service was created (`ModerationService`) to provide a clean interface and to comply with the architectural design
- A new method was added to the `NetworkService` to make the actual POST request
- Minor refactoring in the `Chat` component

### CODE REVIEW

File: src/components/Session/Chat.vue [hunk 1]
```diff
@@ -1,79 +1,112 @@
...
-						<div class="avatar" v-bind:style="message.avatarStyle"></div>
-						<div class="contents">
-              <div class="name">
-								{{message.name}}
-							</div>
-							{{message.contents}}
-              <div class="time">
-								{{message.time}}
-							</div>
+<div class="chat">
```
Comment (stackptr) [inline src/components/Session/Chat.vue]: Lot of stylistic changes make this PR very noisy. It's hard to tell what is changed and what is simply reformatted. Please put stylistic changes in their own PRs and avoid those changes in functional PRs

File: src/components/Session/Chat.vue [hunk 2]
```diff
@@ -1,79 +1,112 @@
...
 			currentSession: SessionService.currentSession,
 			newMessage: ''
 		}
 	},
+
   methods: {
-    sendMessage: function() {
-      var message = this.newMessage;
+    showModerationWarning() {
+      document.querySelector('.chat-warning').style.top = 0;
```
Comment (stackptr) [inline src/components/Session/Chat.vue]: You should use `v-show` and set a variable when the chat warning should be shown. Do not use CSS for that.

File: src/components/Session/Chat.vue [hunk 3]
```diff
@@ -1,79 +1,112 @@
...
-        return 'right';
+
+    sendMessage() {
+      let message = this.newMessage.slice(0,-1);
+
+      if (message != '') {
+
+        ModerationService.checkIfMessageIsClean(this, message).then((isClean) => {
+
+          if (isClean != null) {
```
Comment (stackptr) [inline src/components/Session/Chat.vue]: Logic is:

` if (isClean === true) { this.showNewMessage(message) } else {this.showModerationWarning()}`

Also use strict equality

File: src/components/Session/Chat.vue (hunk 1 above)
Comment (lualparedes) [reply src/components/Session/Chat.vue]: I'll keep this in mind for future PRs, though the weird spacing is caused by a mix of tabs and spaces.

I'm going to convert all tabs to spaces and push, it looks more readable in the standardized version :smile:.

File: src/components/Session/Chat.vue (hunk 3 above)
Comment (lualparedes) [reply src/components/Session/Chat.vue]: @stackptr This was also what I thought in the beginning, but then I realized that this logic doesn't work because if for some reason the cleanliness of the message can't be determined, then `isClean` is going to be `null` (see [this line](https://github.com/UPchieve/web/pull/72/files#diff-4fbc7cb1c3b6d337eeaf0fd91c9284fbR6)), and then the user won't be able to send any message (since `isClean` will keep being `null`, ergo different than `true`), which is undesirable because it breaks the UX completely.

A valid option to avoid using two conditionals would be:
`if (isClean === false) { this.showModerationWarning(); } else { this.showNewMessage(message); }`

Would you rather I use this option or leave it as it is?

Comment (stackptr) [reply src/components/Session/Chat.vue]: Since we control the `ModerationService` method and the desired logic is to display the message in an event of API error, it is clearer if the method itself simply returned `true`. Ideally we will eventually want this to throw an error that can be caught and handled appropriately.

**Comment by lualparedes** [comment]
@stackptr :wave: I just submitted the last requested change :v:

**Comment by alymurray** [comment]
@lualparedes could you address the conflicts in this branch and then I will merge in? Sorry it took me a while to get to this branch.
